pmlc/target/intel_gen/pipeline.cc

```diff
@@ -16,43 +19,52 @@
 #include "pmlc/dialect/tile/transforms/passes.h"
 
 using namespace mlir; // NOLINT[build/namespaces]
-using pmlc::conversion::pxa_to_affine::createLowerPXAToAffinePass;
 
 namespace pmlc::target::intel_gen {
 
 namespace {
 
 void addToPipeline(OpPassManager &pm) {
-  pm.addPass(pmlc::dialect::tile::createComputeBoundsPass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(dialect::tile::createComputeBoundsPass());
+  // pm.addPass(dialect::tile::createPadPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
-  pm.addPass(pmlc::conversion::tile_to_pxa::createLowerTileToPXAPass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(conversion::tile_to_pxa::createLowerTileToPXAPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
   // TODO: do optimizations here
 
-  pm.addPass(createLowerPXAToAffinePass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
-
+  // pm.addPass(std::make_unique<UpdateWorkGroupSizePass>(workGroupSize));
+  // pm.addPass(std::make_unique<IREETileLinalgPass>());
+  // pm.addPass(createConvertLinalgToLoopsPass());
+  pm.addPass(conversion::pxa_to_affine::createLowerPXAToAffinePass());
   pm.addPass(createLowerAffinePass());
-  pm.addNestedPass<FuncOp>(createCanonicalizerPass());
-  pm.addNestedPass<FuncOp>(createCSEPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
 
+  // pm.addPass(std::make_unique<LoopsToGPUPass>());
   pm.addPass(createSimpleLoopsToGPUPass(1, 1));
```

Comment:
@YangleiZouIntel  This pass should be converting outer loops into `gpu.func`

Reply:
Thanks. It seems each outermost loop.for will be translated into a single gpu module. If the outermost loops have the same index and range, we can merge them so it will generate a single  gpu module with a single function.

Reply:
We have a fusion pass that operates on the affine dialect which would be used to merge compatible loops.

What we need in this situation is to either:
1. fix the upstream code so that it only ever creates a single `gpu.module`
2. write a new pass the merges all `gpu.module`s into a single one

Reply:
Thanks. I see. I agree it's better to merge compatible loops in fusion pass or as you mentioned, we can fix the existing outlining pass to generate single gpu.module.